src/otp-verification/otp-verification.tsx

```diff
@@ -0,0 +1,122 @@
+
+export const OtpVerification = (props: OtpVerificationProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        id,
+        "data-testid": dataTestId,
+        onSendOtp,
+        onVerifyOtp,
+        verifyOtpCountdownTimer = 60,
+        sendOtpError,
+        verifyOtpError,
+        otpValue,
+        onOtpChange,
+    } = props;
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [isVerifyLoading, setIsVerifyLoading] = useState(false);
+    const [internalState, setInternalState] = useState<InternalOtpState>(
+        InternalOtpState.DEFAULT
+    );
+
+    const countdown = useCountdown({
+        duration: verifyOtpCountdownTimer,
+    });
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    // Send OTP
+    const handleSendOtp = async () => {
+        if (!onSendOtp) return;
+
+        try {
+            setIsLoading(true);
+            await onSendOtp();
+            setInternalState(InternalOtpState.SENT);
+            // Reset the countdown
+            countdown.reset();
+            // Start the countdown
+            countdown.start();
+        } catch (error) {
+            // do nothing
+        } finally {
+            setIsLoading(false);
+        }
+    };
+
+    // Verify OTP
+    const handleVerifyOtp = async () => {
+        if (!onVerifyOtp || !otpValue?.value) return;
+
+        try {
+            setIsVerifyLoading(true);
+            await onVerifyOtp(otpValue?.value);
+            setInternalState(InternalOtpState.VERIFIED);
+            // Reset the countdown and clear the OTP code in case user enter new OTP later
+            countdown.reset();
+            onOtpChange?.("");
+        } catch (error) {
+            // do nothing
+        } finally {
+            setIsVerifyLoading(false);
+        }
+    };
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const resetToDefault = () => {
+        if (internalState === InternalOtpState.VERIFIED) {
+            setInternalState(InternalOtpState.DEFAULT);
+        }
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    return (
+        <OTPInputWrapper
+            id={id}
+            data-testid={dataTestId}
+            role="region"
+            aria-label="OTP verification"
+        >
+            <ContactInputSection
+                {...props}
+                id={id ? `${id}-contact` : undefined}
```

Comment:
this field needs to be properly associated with the label for accessibility

on further thought, introduce a new `inputId` prop that will be passed to `OtpVerification > ContactInputSection > EmailContactInput/PhoneContactInput` without any modifications

FormOtpVerification should pass this as
```
inputId={`${inputId}-base`}
id={`${inputId}-base-field`} // pick a unique id
```